spark/src/main/java/com/netflix/iceberg/spark/source/IcebergSource.java

```diff
@@ -136,4 +90,31 @@ private static void mergeIcebergHadoopConfs(
         .filter(key -> key.startsWith("iceberg.hadoop"))
         .forEach(key -> baseConf.set(key.replaceFirst("iceberg.hadoop", ""), options.get(key)));
   }
+
+  @Override
+  public org.apache.spark.sql.sources.v2.Table getTable(DataSourceOptions options) {
+    Configuration conf = new Configuration(lazyBaseConf());
+    Table table = getTableAndResolveHadoopConfiguration(options, conf);
+    return new IcebergSparkTable(table);
+  }
+
+  @Override
+  public org.apache.spark.sql.sources.v2.Table getTable(DataSourceOptions options, StructType schema) {
```

Comment:
There should be no need to implement this method. Iceberg tables always have a schema so it makes no sense to supply one. That's for cases like CSV where a schema is required to interpret the data (column names, and types). Otherwise, normal projection works just fine.

Reply:
Don't you still need it for write? Sort of analogous to the previous `getWriter()` method? Or does DSV2 expect that the table will already be created since the datasource layer is separate from the catalog layer, which includes table and schema information?

Reply:
In DSv2, the writer is passed the physical plan and schema, then calls TableCatalog.createTable with that schema. After it creates the table, it uses the instance returned by create.

These methods are called by the DataFrameReader and DataFrameWriter and assume that the table already exists.